### ISSUE TITLE
[Bug]:  Netflix taking forever to load

### ISSUE THREAD

**Issue by timnew** [immersive-translate/immersive-translate#1886]
### 插件版本号 | Version

1.7.5

### 平台 | Platform

macOS

### 浏览器 | Browsers

Arc 1.49.1 

### 插件类型 | Extension Type

浏览器插件 | Browser Extension

### 请描述这个Bug | Describe the bug

The video takes forever to load, and there is no active network activity.
Seems the plugin missed the finish signal or so.

Here is the screen recording of what is going on:
It is in WebM format, modern browser should support to play it
[CleanShot 2024-07-11 at 14.18.13.webm](https://github.com/immersive-translate/immersive-translate/assets/1287238/ff14d9d1-0a43-4ae3-8262-cf99c6b85e56)

### 出现问题的网址 | URL

https://www.netflix.com/watch/81513769

### 重现步骤 | To Reproduce

1. Set the translation service to Claude or OAI, both will end with same result.
2. load the video on Netflix, and it will stuck on loading screen says plugin is translating the subtitles.

### 补充说明 | Additional context

You can see I checked the the network activity that is going out from the browser, and I can see the subtitles are pulled down as xml files. 
There are a few of them, and I can see the last subtitle file was surely pulled, which is talking about the next episode or so, hints it is the ending of all the subtitles.

After that, no new download would happen, except the Netflix logging and push notification. By then I would expect the translation have been done. but the video will never start.

I tried keep the tab opened for 30min long, and it is still stuck on loading screen.

And unfortunately I cannot find the traffic that goes to either OAI or Claude API, which I cannot determine is it a API side of issue. But I tried Claude, OAI, AzureOAI, with different models, none of them works. So I assume it is more an extension issue rather than the API issue. 

And BTW, the google translation works.

**Comment by meijin1123** [comment]
I set the translation service to Claude and use Arc version is1.51.0 (51691)，the https://www.netflix.com/watch/81513769
 translate is OK，you can upgrade your Arc  and try it.

**Comment by timnew** [comment]
Nah, surely not the ARC version reason, it has been quite a while, and I surely upgraded the browser a few times.

And I'm on 1.51.0 (51691)
![CleanShot 2024-07-16 at 12 27 37@2x](https://github.com/user-attachments/assets/13a57164-b4fc-4e0a-ad42-1ad221847aba)

![CleanShot 2024-07-16 at 12 26 45@2x](https://github.com/user-attachments/assets/9552f9e1-68cf-4c16-9230-f10a367f5e1b)

**Comment by meijin1123** [comment]
![image](https://github.com/user-attachments/assets/0eec0af4-6175-41bd-b717-ff9c085ea2de)
![image](https://github.com/user-attachments/assets/5fcfc10b-eb26-482b-8b6c-5d4e089c35fc)
![image](https://github.com/user-attachments/assets/937f9647-39d3-401e-be5e-66bebb3be731)

----I just checked the results， you can try restarting the plugin(orClear the cache）

**Comment by timnew** [comment]
Here are a few things that I have tried:

1. Tried on a clean Chrome, with no other extension installed. To avoid ARC compatibility issue and potential interferences from other extensions, such as AdBlocker or so. **Same issue.**
2. Tried on title in other languages, videos from other country, **still same issue.**
3. Tried on other platform, such as Youtube, **it works fine.** 
![CleanShot 2024-07-16 at 12 40 21@2x](https://github.com/user-attachments/assets/db550674-56f3-4125-84d8-940301eba626)
4. Tried OAI and Claude, both doesn't work, same issue.
5. Tried use OAI compatible API via a reversed proxy, bridged to local LLM or an Azure OAI deployment, same issue.
6. Tried Google Translation, it works fine.

So my guess the issue is some how related to LLM and Netflix, but browser related, environment related or API related.

**Comment by keyboard3** [comment]
Netflix subtitle translation does not recommend using a slow response translation service, because in order to restore the subtitle effect, its subtitle request is displayed only after all subtitles have been translated, so once you use openAI or Cluade, a slow translation service, it will wait for a very old one. And YouTube doesn't translate progressively because it's an immersive self-hosted subtitle

**Comment by timnew** [comment]
Translation wasn't shown, even though the extension seems being active.
![CleanShot 2024-07-16 at 12 54 19@2x](https://github.com/user-attachments/assets/ffca5731-47fb-4409-bc8a-5ee512c345b1)

Translation was shown, but the extension icon weren't loaded
![CleanShot 2024-07-16 at 12 52 22@2x](https://github.com/user-attachments/assets/a86314bb-fd52-4f85-819d-432653de5040)

**Comment by timnew** [comment]
@keyboard3 Why closing this issue? It is still being actively discussed?
Issue still exists and unsolved. I'm still trying to provide more information.
And why you just close it?

**Comment by keyboard3** [comment]
As I said above, Netflix subtitles should be translated with a responsive translation service, such as Google. You shouldn't use slow-response translation services like openAI or cluade. Because it waits for all the subtitles to be translated before it can be displayed

**Comment by timnew** [comment]
@keyboard3 I'm okay if you don't want to help on this issue, but don't close it, as I need other's help if they are willing to or share the same issue.

Here is not slow or not slow issue. As I mentioned before, I left it there for half an hour, nothing happened. Nothing would take half an hour to download/translate. 

Also I mentioned, it is clearly all the subtitles are downloaded, it took quite a few seconds to finish, but I can see the traffic was all downloaded, but video doesn't start.

From my experience, it works on some of the video, but not the others. Same netflix, same extension, same translation service. So I can't say it is the issue about the speed or so.

For speed of LLM, I even bridged the LLM to groq and local small LLM, both should be quick, but none of them works.

So I'm buying this is issue of mater of LLM speed or so.

The reason I'm opening an issue here is to ask for help, openly. So people shares similar situation can help each other. Otherwise I can investigate on this on my own, or try something else.

If you don't want to help on this issue, that's fine, but don't close this issue, as I need help from others, and others can save some time on experiment if they have the same issue.

**Comment by timnew** [comment]
@keyboard3 thanks mate.

**Comment by FattySpring** [comment]
Netflix pre-translated all subtitles have been changed to lazy loading based on translation progress, but you can't use Netflix's original style.
